core/src/main/java/org/apache/iceberg/ManifestLists.java

```diff
@@ -43,22 +47,54 @@ private ManifestLists() {
         .reuseContainers(false)
         .build()) {
 
-      return Lists.newLinkedList(files);
+      return shouldUseRelativePaths ? updateManifestFilePathsToAbsolutePaths(files,
```

Comment:
I believe this was commented on the issue as well, but shouldn't we always call updateManifestFilePathstoAbsolute here? Instead of passing through a boolean we could just see if we need to update any files and return the updated set? So a use with only absolute paths would essentially just pass through the list and not change anything?

This would remove our requirement to pass through "shouldUseRelativePaths"